code/modules/mob/living/silicon/robot/drone/drone.dm

```diff
@@ -198,8 +194,7 @@
 	emagged = 1
 	density = 1
 	pass_flags = 0
-	icon_state = "repairbot-emagged"
-	holder_type = /obj/item/holder/drone/emagged
+	icon_state = "[icon_state]-emag" //applies emag-specific sprites based on current chassis
```

Comment:
Emagging a drone twice won't result in an invalid icon_state?

Reply:
emag_act() returns before we get to the icon state reassignment if the drone is already emagged, unless i'm missing something